crosstool/osx_cc_configure.bzl

```diff
@@ -88,6 +88,26 @@ def _generate_system_modulemap(repository_ctx, script, output):
 def _copy_file(repository_ctx, src, dest):
     repository_ctx.file(dest, content = repository_ctx.read(src))
 
+def _get_tool_path(repository_ctx, tool_name, path_env_var):
+    tool_path = repository_ctx.os.environ.get(path_env_var)
+    if tool_path != None:
+        if not tool_path.startswith("/"):
+            tool_path = repository_ctx.which(tool_path)
+        return tool_path
+    else:
+        env = repository_ctx.os.environ
+        result = repository_ctx.execute([
+            "env",
+            "-i",
+            "DEVELOPER_DIR={}".format(env.get("DEVELOPER_DIR", default = "")),
+            "xcrun",
+            "--find",
+            tool_name,
+        ])
+        if result.return_code != 0:
+            return None
+        return result.stdout.strip()
```

Comment:
does these being host specific absolute paths affect caching? since this will have the absolute path to Xcode.app in it

Reply:
Yea i think that would work.

I think the reason the gcov one has flown under the radar is because in apple specific (aka non-c++) tests the coverage logic is handled in rules_apple and sidesteps all of this

Reply:
no worries. I think the updated version addresses this issue. If no overrides are present then the cache key contains a script which just says `xcrun --run gcov/llvm-profdata/llvm-cov`. Since the script is the same across machines caching should work fine.
If there are override env vars present, then as long as users set the override to the same value, they should also get cache hits.